feat: add reverse_sort input option for sorting functionality

kakkoyun opened the pull request:
This PR implements a `sort_order` parameter to provide more flexible control over the display order of blog posts. This addresses maintainer feedback on a previous iteration that used a boolean `reverse_sort` flag, and incorporates modifications to the initial suggestion.

Key changes:
- Added `sort_order` input to `action.yml`.
  - Supports `'asc'` (oldest first) and `'desc'` (newest first, default).
  - This parameter only takes effect if date sorting is enabled (i.e., `disable_sort` is `'false'`).
  - If `disable_sort` is `'true'`, `sort_order` has no effect, and posts remain in their fetched order.
- Updated `README.md` to document the new `sort_order` option and its behavior.
- Modified `blog-post-workflow.js` to implement the `sort_order` logic.
  - Posts are sorted directly in the specified order in a single step.
  - Items without publication dates are consistently placed at the end.
- Updated tests (renamed `test/reverse-sort.js` to `test/sort-order-asc.js` and removed `test/disable-sort-asc.js`) to reflect the new parameter and its conditional behavior.

Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our contribution guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature: Adds a `sort_order` parameter to allow sorting posts in ascending (`asc`, the oldest first) or descending (`desc`, the newest first) order when date-based sorting is active.


* **What is the current behavior?** (You can also link to an open issue here)

Currently, the blog post workflow only supports sorting posts by date in descending order (newest first) by default, or disabling sorting entirely via `disable_sort`. There is no built-in option to sort oldest first when date sorting is active.


* **What is the new behavior (if this is a feature change)?**

This PR introduces a `sort_order` input parameter.
- If `disable_sort` is `'false'` (default, date sorting is ON):
    - `sort_order: 'desc'` (default): Sorts posts newest first.
    - `sort_order: 'asc'`: Sorts posts oldest first.
- If `disable_sort` is `'true'` (date sorting is OFF):
    - The `sort_order` parameter has no effect. Posts will be displayed in the order they are fetched from the RSS feed(s).

The feature handles RSS feeds that may not include `pubDate` for all items; items without dates are placed at the end of the sorted list.


* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No. This is a non-breaking change.
- Existing workflows not using the new `sort_order` parameter will continue to function with the default behavior (newest first if sorting is enabled, or fetched order if sorting is disabled).
- Users can be opt-in to ascending sort by setting `sort_order: 'asc'` when `disable_sort` is `false`.


* **Other information**:

This implementation is the result of discussions following maintainer feedback. It aims to provide the requested sorting flexibility while ensuring clarity and efficiency. The sorting is done in a single pass, directly into the desired order, when date sorting is active.

gautamkrishnar commented on action.yml:
```diff
@@ -23,6 +23,10 @@ inputs:
     description: "Disables the sorting of list based on publish date"
     default: "false"
     required: false
+  reverse_sort:
```
Rename to `sort_order` with default value `forward`. 

Values supported:
- forward
- reverse

kakkoyun replied:
Addressed in b272c3e24a68592b58c8bd492fe357bd65295f6b

gautamkrishnar commented on src/blog-post-workflow.js:
```diff
@@ -140,7 +143,7 @@ feedList.forEach((siteUrl) => {
               if (ENABLE_SORT && ENABLE_VALIDATION && !item.pubDate) {
                 reject('Cannot read response->item->pubDate');
               }
-              
```
revert unnecessary style changes like this

kakkoyun replied:
I understand your gut reaction, but these are mere white space deletions. These shouldn't be here in the first place.

gautamkrishnar commented on src/blog-post-workflow.js:
```diff
@@ -150,11 +153,11 @@ feedList.forEach((siteUrl) => {
                 }
                 // Use URL as fallback or a default text
-                item.title = item.link ? 
-                  `[No Title] - ${item.link.split('/').pop() || 'Post'}` : 
```
revert

gautamkrishnar commented on src/blog-post-workflow.js:
```diff
@@ -150,11 +153,11 @@ feedList.forEach((siteUrl) => {
+                  `[No Title] - ${item.link.split('/').pop() || 'Post'}` :
                   'Post without title';
               }
-              
```
revert

kakkoyun replied:
Same as above https://github.com/gautamkrishnar/blog-post-workflow/pull/281/files#r2083420922

gautamkrishnar reviewed (changes requested):
Please revert all the unecessary changes and make the code changes required

kakkoyun commented:
@gautamkrishnar I have addressed your comments. I have used `asc` and `desc` instead of your suggestion. I believe users would find it more intuitive.

Furthermore, I would rather not bike shed on the whitespace changes, so please let me know if you feel strong about them. I'll revert.

gautamkrishnar commented:
@kakkoyun looks good, thanks a lot for contributing 🌮

gautamkrishnar commented:
@all-contributors please add @kakkoyun for code
